pkg/localmetrics/localmetrics.go

```diff
@@ -134,6 +139,11 @@ func NewMetricsCollector(store cache.Cache) *MetricsCollector {
 			ConstLabels: prometheus.Labels{"name": operatorName},
 			Buckets:     []float64{0.001, 0.01, 0.1, 1, 10, 30, 60, 120, 240},
 		}, []string{"controller"}),
+
+		apiCallCount: prometheus.NewCounterVec(prometheus.CounterOpts{
+			Name: "aws_account_operator_api_requests_total",
```

Comment:
Note that this is called "api requests" and not "kubeclient requests" because we also intend to [accumulate AWS API calls](https://issues.redhat.com/browse/OSD-4507) in the same metric, as noted in the commit message.

Reply:
@2uasimojo looking at the dimensions we'll be able to determine if its a kube API request by the `resource` ?

Reply:
Yes. More below.

Reply:
Saw it after :+1:

Reply:
@2uasimojo Do we want to use this metric for both k8s and AWS API calls? Can I ask why is the design? IMO it is better to separate the metric for AWS and kube.

Reply:
@yeya24 Yes, the intent will be to use the same metric for both.

Scrape-/graph-wise, it's [sixes](https://en.wiktionary.org/wiki/six_of_one,_half_a_dozen_of_the_other): As written, we can filter by resource regex if we want to split them; or if we split them here we can always aggregate them on the graph side.

So it came down to how it's coded. I preferred the idea of funneling everything through a single common method. Also, the metrics for both types of call would be identical in structure, and I would prefer not to repeat.

Reply:
@2uasimojo For AWS API calls, how to get `method`, `status` and `resource`? Do you want to instrument the AWS client?

Reply:
I looked into it *briefly*, and the AWS clients allow you to pass in [handlers](https://github.com/aws/aws-sdk-go/blob/master/aws/request/handlers.go) as part of the config. AFAICT a "handler" contains a [callback function](https://github.com/aws/aws-sdk-go/blob/b868519ad5e70e7b66b3ef4753590fa752996c99/aws/request/handlers.go#L140) to which the [Request](https://github.com/aws/aws-sdk-go/blob/b868519ad5e70e7b66b3ef4753590fa752996c99/aws/request/request.go#L46) is passed, and the Request contains the [http.Request and Response](https://github.com/aws/aws-sdk-go/blob/b868519ad5e70e7b66b3ef4753590fa752996c99/aws/request/request.go#L55-L56).

Hopefully that's as simple as it looks, and we just have to decide a) where to stuff our handler, and b) what we want the `resource` to look like for AWS calls, so we can slice and dice the Request accordingly.